### PR TITLE
V2 lar model

### LINKED ISSUE #1443: Create data model for Loan Application Register
Data spec in the [Filing Instruction Guide](https://www.consumerfinance.gov/data-research/hmda/static/for-filers/2018/2018-HMDA-FIG.pdf)

### PR DESCRIPTION
**Pull request by jmarin**
Closes #1443

### CODE REVIEW

File: src/main/scala/hmda/model/filing/lar/enums/RaceObservedEnum.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+case object VisualOrSurnameRace extends RaceObservedEnum {
+  override val code: Int = 1
+  override val description: String =
+    "Collected on the basis of visual observation or surname"
+}
+
+case object NotVisualOrSurnameRace extends RaceObservedEnum {
+  override val code: Int = 2
+  override val description: String =
+    "Not collected on the bassis of visual observation or surname"
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/enums/RaceObservedEnum.scala]: Typo here

File: src/main/scala/hmda/model/filing/lar/Applicant.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    creditScoreType: CreditScoreEnum,
+    debtToIncomeRatio: Option[String] = None,
+    otherCreditScoreModel: String
+) extends PipeDelimited {
+  override def toCSV: String = {
+    s"${ethnicity.ethnicity1.code}|${ethnicity.ethnicity2.code}|${ethnicity.ethnicity3.code}|${ethnicity.ethnicity4.code}|" +
+      s"${ethnicity.ethnicity5.code}|$otherHispanicOrLatino|${ethnicityObserved.code}|${race.race1.code}|${race.race2.code}|" +
+      s"${race.race3.code}|${race.race4.code}|${race.race5.code}|${raceObserved.code}|$otherNativeRace|$otherAsianRace|" +
+      s"$otherPacificIslanderRace|${sex.code}|${coSex.code}|${sexObserved.code}|${coSexObserved.code}|$age|$coAge|" +
+      s"$income|$creditScore|${creditScoreType.code}|$debtToIncomeRatio|$otherCreditScoreModel"
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/Applicant.scala]: `debtToIncomeRatio` needs to be converted from an Option to a String

File: src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    val businessOrCommercialStr = businessOrCommercialPurpose match {
+      case Some(businessOrCommercial) => businessOrCommercial.code
+      case None                       => ""
+    }
+
+    s"$id|$leiStr|${loan.toCSV}|${preapproval.code}|${actionTakenType.code}|$actionTakenDate|${geography.toCSV}|" +
+      s"${applicant.toCSV}|${purchaserType.code}|$rateSpread|${hoepaStatus.code}|${lienStatus.code}|${denial.toCSV}|" +
+      s"$otherDenialReason|${loanDisclosure.toCSV}|$interestRateStr|$prepaymentPenaltyTerm|$loanToValueRatioStr|$introductoryRatePeriod|" +
+      s"${nonAmortizingFeatures.toCSV}" +
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala]: This field is missing an end pipe

Comment (jmarin) [reply src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala]: What do you mean? I'm not sure I follow

File: src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    val multiFamilyStr = multiFamilyAffordableUnits match {
+      case Some(multifamily) => multifamily
+      case None              => ""
+    }
+
+    val ausStr = AUS match {
+      case Some(aus) => aus.toCSV
+      case None      => ""
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala]: Both here and at line 96 need to account for the missing pipes.  A `None` case should return some set of empty pipes (e.g. `"||||"`).

File: src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      case Some(credit) => credit.code
+      case None         => ""
+    }
+
+    val businessOrCommercialStr = businessOrCommercialPurpose match {
+      case Some(businessOrCommercial) => businessOrCommercial.code
+      case None                       => ""
+    }
+
+    s"$id|$leiStr|${loan.toCSV}|${preapproval.code}|${actionTakenType.code}|$actionTakenDate|${geography.toCSV}|" +
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala]: Preapprovals are at the wrong index.  They should be at position 7, between loan purpose and construction method.

Comment (nickgrippin) [reply src/main/scala/hmda/model/filing/lar/LoanApplicationRegister.scala]: Loan term is also in the wrong position, it should be at position 82

File: src/main/scala/hmda/model/filing/lar/Applicant.scala [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    coAge: Int,
+    income: String,
+    creditScore: Int,
+    creditScoreType: CreditScoreEnum,
+    debtToIncomeRatio: Option[String] = None,
+    otherCreditScoreModel: String
+) extends PipeDelimited {
+  override def toCSV: String = {
+    s"${ethnicity.ethnicity1.code}|${ethnicity.ethnicity2.code}|${ethnicity.ethnicity3.code}|${ethnicity.ethnicity4.code}|" +
+      s"${ethnicity.ethnicity5.code}|$otherHispanicOrLatino|${ethnicityObserved.code}|${race.race1.code}|${race.race2.code}|" +
```
Comment (nickgrippin) [inline src/main/scala/hmda/model/filing/lar/Applicant.scala]: Observed ethnicity is in the wrong position.

Also missing observed co-ethnicity.

**Pull request merged**